components/sup/src/manager/service/hooks.rs

```diff
@@ -418,14 +375,49 @@ impl Hook for PostStopHook {
     fn stderr_log_path(&self) -> &Path { &self.stderr_log_path }
 }
 
+#[derive(Clone, Copy, Debug, Default, Serialize)]
+pub struct HookChangeTable {
+    pub health_check: bool,
+    pub init:         bool,
+    pub file_updated: bool,
+    pub reconfigure:  bool,
+    pub suitability:  bool,
+    pub run:          bool,
+    pub post_run:     bool,
+    pub post_stop:    bool,
+}
+
+impl HookChangeTable {
+    pub fn new() -> Self {
+        Self { health_check: false,
+               init:         false,
+               file_updated: false,
+               reconfigure:  false,
+               suitability:  false,
+               run:          false,
+               post_run:     false,
+               post_stop:    false, }
+    }
+
+    pub fn changed(self) -> bool {
+        self.health_check
+        || self.init
+        || self.file_updated
+        || self.reconfigure
+        || self.suitability
+        || self.run
+        || self.post_run
+        || self.post_stop
+    }
```

Comment:
Another alternative would be using a `HashMap` wrapper with an array of constant associated keys instead of a struct with separate fields. That would allow us to do something like `self.0.values().any(…)`.